app/src/main/java/unipd/se18/ocrcamera/GalleryActivity.java

```diff
@@ -0,0 +1,385 @@
+            ActionBar actionBar =((GalleryActivity)getActivity()).getSupportActionBar();
+            actionBar.setTitle(R.string.galleryFragmentTitle);
+            actionBar.setDisplayHomeAsUpEnabled(false);
+            actionBar.setHomeButtonEnabled(false);
+        }
+
+        /**
+         * Event triggered when I create the view
+         */
+        @Override
+        public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState)
+        {
+            // Defines the xml file for the fragment
+            return inflater.inflate(R.layout.activity_gallery_content, parent, false);
+        }
+
+        /**
+         * This event is triggered soon after onCreateView().
+         * Any view setup should occur here.
+         */
+        @Override
+        public void onViewCreated(View view, Bundle savedInstanceState)
+        {
+            //Create the container for all my cards
+            RecyclerView picturesRecycleView = view.findViewById(R.id.recycle_view);
+            picturesRecycleView.setHasFixedSize(true);
+
+            //Set a grid (2 columns) from my recycleView
+            GridLayoutManager recycleLayoutManager = new GridLayoutManager(view.getContext(), 2, GridLayoutManager.VERTICAL, false);
+            picturesRecycleView.setLayoutManager(recycleLayoutManager);
+            picturesRecycleView.setItemAnimator(new DefaultItemAnimator());
+
+            //Load the cards using the RecycleCardsAdapter into the picturesRecycleView
+            ArrayList<GalleryManager.PhotoStructure> photos = GalleryManager.getImages(view.getContext());
+            cardAdapter = new GalleryManager.RecycleCardsAdapter(view.getContext(), photos);
+            picturesRecycleView.setAdapter(cardAdapter);
+        }
+
+
+        /**
+         * Event when the main fragment change its hidden status.
+         * @param hidden true=is hidden, false=is not hidden.
+         */
+        @Override
+        public void onHiddenChanged(boolean hidden)
+        {
+            //I have to understand if the user has deleted an image
+            //If the answer is yes I have to remove the item from the recycleView
+            if(!hidden)
+            {
+                //When I delete something from the DetailsFragment using the method GalleryManager.deleteImage(photoInfos)
+                //I set in the GalleryManager an object containing the PhotoStructure to delete.
+                //Calling .getDeletedPhotoPosition() I check if the value is -1, if not I get the position of
+                //the PhotoStructure to delete
+
+                int deletedPosition=cardAdapter.getDeletedPhotoPosition();
+                if(deletedPosition!=-1)
+                {
+                    //have to remove the photo from the recycler
+                    cardAdapter.photosList.remove(deletedPosition);
+                    cardAdapter.notifyItemRemoved(deletedPosition);
+                }
+
+                //Restore the actionBar
```

Comment:
The lines of code from the line 166 to the line 169, the lines of code from the line 191 to the line 194 and the lines of code from the line 102 to the line 105 are the same.
If they don't specialize this fragment only for the use with the GalleryActivity, could a good idea be putting them to a single method that is launched every times it's needed?

Reply:
They are not the same because in I show/hide the activity's buttons. If I want to create a method I need to create it with 3 parameters so I opted for this method because I also get a more clear way to understand the settings of the actionbar are in each fragment